src/coreclr/gc/gc.cpp

```diff
@@ -36496,48 +36655,63 @@ size_t gc_heap::approximate_new_allocation()
     return max (2*dd_min_size (dd0), ((dd_desired_allocation (dd0)*2)/3));
 }
 
+bool gc_heap::check_against_hard_limit (size_t space_required)
+{
+    bool can_fit = TRUE;
+
+    // If hard limit is specified, and if we attributed all that's left in commit to the ephemeral seg
+    // so we treat that as segment end, do we have enough space.
+    if (heap_hard_limit)
+    {
+        size_t left_in_commit = heap_hard_limit - current_total_committed;
+        int num_heaps = 1;
+#ifdef MULTIPLE_HEAPS
+        num_heaps = n_heaps;
```

Comment:
I am assuming the compiler will optimize this, but wondering if `num_heaps` should be initialized with `n_heaps` rather than assigning twice?

Reply:
n_heaps is not defined when MULTIPLE_HEAPS is not defined

Reply:
I meant something like:
```
#ifdef MULTIPLE_HEAPS
        int num_heaps = n_heaps;
#else
        int num_heaps = 1;
#endif //MULTIPLE_HEAPS
```

Reply:
I can do that but I don't think you will be able to detect any perf difference at all, even if the compiler didn't optimize it (it probably does) :)

Reply:
fair enough..